### PR TITLE
Adds a regular expression for device naming for #50 issue

### PR DESCRIPTION
**Pull request by TheSkorm**
This adds validation (like attributes) for hostnames which should resolve issue #50

### CODE REVIEW

**Comment by jathanism** [comment]
Hi there. Thanks for the patch! Can you please make the pattern inclusive to allow for valid hostname characters? As you can see, your patch broke the unit tests that are using hostnames like `foo-bar1` (where a hyphen is a valid hostname character).

If you could fix the pattern, and also implement a couple minor unit tests for it exercising the hostname validation, that would be excellent! Thanks!

**Comment by TheSkorm** [comment]
Will do. Didn't see the unit tests, will have a look

**Comment by TheSkorm** [comment]
Regexp has been updated and added unit tests.

**Comment by jathanism** [comment]
Awesome! So, last steps:

- Increment the version in `nsot/version.py` to `0.14.1`
- Rebase everything to a single commit!

Thanks for your contribution!

File: tests/api_tests/test_devices.py [hunk 1]
```diff
@@ -34,6 +34,15 @@ def test_creation(live_server, user, site):
 
     admin_client.create(attr_uri, resource_name='Device', name='attr1')
 
+    # Test invalid attribute name
```
Comment (jathanism) [inline tests/api_tests/test_devices.py]: *hostname (not attribute name)

**Comment by TheSkorm** [comment]
Rebased and fixed up that comment (I wonder where I borrowed that code from ;) ). Version number also updated.

Thanks for guiding me through the pull process. Looking forward to working on some other bugs.

**Comment by jathanism** [comment]
@TheSkorm My pleasure. And thanks again for your contribution. Please consider joining us at #nsot on IRC (Freenode).

**Pull request merged**